iphone/Classes/TiUIWindowProxy.m

```diff
@@ -920,6 +924,86 @@ -(void)setToolbar:(id)items withObject:(id)properties
 	
 }
 
+-(void)setHidesBarsOnSwipe:(id)value
+{
+    ENSURE_TYPE(value, NSNumber);
+
+    [self replaceValue:value forKey:@"hidesBarsOnSwipe" notification:NO];
+    [self updateHidesBarsOnSwipe];
+}
+
+-(void)setHidesBarsOnTap:(id)value
+{
+    ENSURE_TYPE(value, NSNumber);
+
+    [self replaceValue:value forKey:@"hidesBarsOnTap" notification:NO];
+    [self updateHidesBarsOnTap];
+}
+
+-(void)setHidesBarsWhenVerticallyCompact:(id)value
+{
+    ENSURE_TYPE(value, NSNumber);
+
+    [self replaceValue:value forKey:@"hidesBarsWhenVerticallyCompact" notification:NO];
+    [self updateHidesBarsWhenVerticallyCompact];
+}
+
+-(void)setHidesBarsWhenKeyboardAppears:(id)value
+{
+    ENSURE_TYPE(value, NSNumber);
+
+    [self replaceValue:value forKey:@"hidesBarsWhenKeyboardAppears" notification:NO];
+    [self updateHidesBarsWhenKeyboardAppears];
+}
+
+-(void)updateHidesBarsOnSwipe
+{
+    if ([TiUtils isIOS8OrGreater]) {
+        TiThreadPerformOnMainThread(^{
+            if ((controller != nil) && ([controller navigationController] != nil)) {
+                UINavigationController *ourNC = [controller navigationController];
+                ourNC.hidesBarsOnSwipe = [TiUtils boolValue:[self valueForUndefinedKey:@"hidesBarsOnSwipe"] def:NO];
+            }
+        }, NO);
+    }
+}
```

Comment:
I would do that in the setters directly to keep the number of methods small and the code semantically together.